Fix test PT-1966

yoku0825 opened the pull request:
- [\[PT\-1966\] t/pt\-online\-schema\-change/no\_drop\_no\_swap\.t is broken \- Percona JIRA](https://jira.percona.com/browse/PT-1966)

no_drop_no_swap.t  dies by `Error altering new table `pt_osc`.`_t_new`: DBD::mysql::db do failed: Duplicate column name 'd' [for Statement "ALTER TABLE `pt_osc`.`_t_new` ADD COLUMN d INT"]` .

This pull-request changes adding column name to avoid duplicate column name.

yoku0825 committed:
Detect raising error for PT-1966

yoku0825 committed:
Fix non-duplicated column name for PT-1966

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/signed)](https://cla-assistant.percona.com:443/percona/percona-toolkit?pullRequest=493) <br/>All committers have signed the CLA.

percona-csalguero reviewed (changes requested):
Hi, thanks for the contribution.
Could you please change the test removing the `eval` and checking the exit code like we do  [here](https://github.com/percona/percona-toolkit/blob/3.x/t/pt-online-schema-change/basics.t#L56-L59) ?
```
($output, $exit) = full_output(
   sub { pt_online_schema_change::main(@args, "$dsn,D=pt_osc,t=t",
      '--alter', 'drop column id') }
);
```

yoku0825 committed:
More good way to catch an error. (PT-1966)

- https://github.com/percona/percona-toolkit/pull/493#pullrequestreview-678426876

yoku0825 commented:
@percona-csalguero Hi! Thank you for your suggestion, that's a good way to catch an error!
I fixied and push it. thanks!

percona-csalguero commented:
One more thing, please.
Could you update the changelog and mention youself as the contributor? (if you want to)
Thanks

yoku0825 committed:
Add Changelog (PT-1966)

- https://github.com/percona/percona-toolkit/pull/493

yoku0825 commented:
I added `Changelog`, thank you.

percona-csalguero commented:
Hi, There was another pull request and now the Changelog has a conflict. 
Could you please fix it, leaving both changes 1965 and 1966?
Once it is fixed, I'll merge it.
Thanks

yoku0825 committed:
Merge branch '3.x' into fix_test-PT1966

yoku0825 commented:
I fixed the conflict. thanks!
